postprocessing_variant_calls/mutect/mutect_class.py

```diff
@@ -0,0 +1,272 @@
+
+        # If the caller reported the normal genotype column before the tumor, swap those around
+        if self.allsamples[1] == self.tsampleName:
+            self.vcf_reader.samples[0] = self.allsamples[1]
+            self.vcf_reader.samples[1] = self.allsamples[0]
+    
+        # Dictionary to store records to keep
+        keepDict = {}
+
+        # Filter each row (Mutation)
+        txtDF = pd.read_table(self.inputTxt, skiprows=1, dtype=str)
+        txt_fh = open(txt_out, "wb")
+        for index, row in txtDF.iterrows():
+            chr = row.loc['contig']
+            pos = row.loc['position']
+            ref_allele = row.loc['ref_allele']
+            alt_allele = row.loc['alt_allele']
+            trd = int(row.loc['t_ref_count'])
+            tad = int(row.loc['t_alt_count'])
+            nrd = int(row.loc['n_ref_count'])
+            nad = int(row.loc['n_alt_count'])
+            judgement = row.loc['judgement']
+            failure_reason = row.loc['failure_reasons']
+
+            tdp,tvf = _tumor_variant_calculation()
+            ndp,ndf = _normal_variant_calculation()
+            nvfRF = _tvf_threshold_calculation()
+
+            # This will help in filtering VCF
+            key_for_tracking = str(chr) + ':' + str(pos) + ':' + str(ref_allele) + ':' + str(alt_allele)
+
+            if judgement != 'KEEP':
+                # Check the failure reasons to determine if we should still consider this variant
+                failure_tags = failure_reason.split(',')
+                tag_count = 0
+                for tag in failure_tags:
+                    if tag in ACCEPTED_TAGS:
+                        tag_count += 1
+                # All failure_reasons should be found in accepted tags to continue
+                if tag_count != len(failure_tags):
+                    continue
+            else:
+                failure_reason = 'KEEP'
+
+            if tvf > nvfRF:
+                if (tdp >= int(args.dp)) & (tad >= int(args.ad)) & (tvf >= float(args.vf)):
+                    if key_for_tracking in keepDict:
+                        print('MutectStdFilter: There is a repeat ', key_for_tracking)
+                    else:
+                        keepDict[key_for_tracking] = failure_reason
+                    out_line = str.encode(args.tsampleName + "\t" + str(chr) + "\t" + str(pos) + "\t" + str(ref_allele) + "\t" + str(alt_allele) + "\t" + str(failure_reason) + "\n")
+                    txt_fh.write(out_line)
+        txt_fh.close()
+
+        # This section uses the keepDict to write all passed mutations to the new VCF file
+        _write_to_vcf(self.vcf_out,self.vcf_reader,self.allsamples,keepDict)
+
+
+        return self.vcf_out, self.txt_out
+
+
+
+def _tumor_variant_calculation(trd,tad):
+    ##############################
+    # Tumor Variant Calculations #
+    ##############################
+
+    tdp = trd + tad
+
+    if tdp != 0:
+        tvf = int(tad) / float(tdp)
```

Comment:
We should also consider writing division error cases. For example, division by zero.